app.go

```diff
@@ -177,7 +177,7 @@ func (a *App) buildInstance() (*registry.ServiceInstance, error) {
 		for _, srv := range a.opts.servers {
 			if r, ok := srv.(transport.Endpointer); ok {
 				e, err := r.Endpoint()
-				if err != nil {
+				if err != nil || e == nil {
```

Comment:
Maybe we should check for `nil` pointers and then directly continue to skip invalid `*url.URL`.